library/cipher.c

```diff
@@ -150,6 +155,7 @@ const mbedtls_cipher_info_t *mbedtls_cipher_info_from_values( const mbedtls_ciph
 
 void mbedtls_cipher_init( mbedtls_cipher_context_t *ctx )
 {
+    CIPHER_VALIDATE( ctx != NULL );
```

Comment:
I think the NULL check in `mbedtls_cipher_info_from_string()` should be moved to be part of parameter validation.